src/app/task-list/task-list.component.ts

```diff
@@ -2,29 +2,60 @@ import { Component, OnInit } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { TaskService } from '../service/task.service';
 import { Task } from '../models/task.model';
-import { RouterLink } from '@angular/router';
+import { Subject, takeUntil, switchMap } from 'rxjs';
+import { Router, RouterLink } from '@angular/router';
+import { error } from 'node:console';
 
 @Component({
   selector: 'tasko-task-list',
   standalone: true,
-  imports: [CommonModule,RouterLink],
+  imports: [CommonModule, RouterLink],
   templateUrl: './task-list.component.html',
   styleUrl: './task-list.component.scss'
 })
 export class TaskListComponent implements OnInit {
   tasks: Task[] = [];
+  private destroy$: Subject<void> = new Subject<void>;
 
-  constructor(private taskService: TaskService) {
-    // console.log(data);
+  constructor(private taskService: TaskService, private router: Router) {
   }
 
   ngOnInit() {
-    
-    this.tasks = this.taskService.getTasks();
+    this.getTasks();
+  }
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
   }
 
-  onDeleteTask(index: number) {
-    this.taskService.deleteTask(index);
+  getTasks() {
+    this.taskService.getAllTasks().pipe(takeUntil(this.destroy$)).subscribe({
+      next: (response: Task[]) => {
+        this.tasks = response;
+      },
+      error: (error: Error) => {
+        console.error(error);
+      }
+    })
   }
+  deleteATask(taskId: string): void {
+
+    this.taskService.deleteATask(taskId)
+
+      .pipe(
+        takeUntil(this.destroy$),
```

Comment:
to avoid doing 2 requests to fetch new list of tasks after deletion, you can return a list of tasks after deleting a task